android/app/src/main/res/layout/activity_search.xml

```diff
@@ -0,0 +1,190 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```

Comment:
I have to see how it looks but I think you should think of organizing your items in ScrollViews/NestedScrollViews (with listview inside) to be scrollable as the results of search should be on the same screen and the second thisnk is to organize the items into groups with the [include/merge](https://developer.android.com/training/improving-layouts/reusing-layouts) (as there are going to be a lot of them)